frontend/web/src/app/stl-viewer/classes/Utilities.ts

```diff
@@ -0,0 +1,21 @@
+export class Util {
+
+	/**
+	 * Convert content of a responseBody, as the result of an XHR request, to a (binary) string.
+	 * This method is special for IE.
+	 */
+	ieXHRResponseBodyToString(responseBody) {
+		// I had expected this could be done by a single line:
+		//     String.fromCharCode.apply(null, (new VBArray(responseBody)).toArray());
+		// But it tends to result in an 'out of stack space' exception on larger streams.
+		// So we just cut the array to smaller pieces (64k for each) and convert them to
+		// strings which can then be combined into one.
+
+		//var arr = (new VBArray(responseBody)).toArray();
+		var arr = responseBody.toArray();	// TODO: WARNING: This might fuck up completely.
+		var str = '';
+		for(var i=0; i<arr.length-65536; i+=65536)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') for statements must be braced [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS121)
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Forbidden 'var' keyword, use 'let' or 'const' instead [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS3504)